src/main/java/graphql/servlet/GraphQLServlet.java

```diff
@@ -338,33 +339,38 @@ private void query(String query, String operationName, Map<String, Object> varia
             final ExecutionResult executionResult = newGraphQL(schema).execute(new ExecutionInput(query, operationName, context, rootObject, transformVariables(schema, query, variables)));
             final List<GraphQLError> errors = executionResult.getErrors();
             final Object data = executionResult.getData();
+            final Object extensions = executionResult.getExtensions();
 
-            final String response = getMapper().writeValueAsString(createResultFromDataAndErrors(data, errors));
+            final String response = getMapper().writeValueAsString(createResultFromDataErrorsAndExtensions(data, errors, extensions));
 
             GraphQLResponse graphQLResponse = new GraphQLResponse();
             graphQLResponse.setStatus(STATUS_OK);
             graphQLResponse.setResponse(response);
             responseHandler.handle(graphQLResponse);
 
             if(getGraphQLErrorHandler().errorsPresent(errors)) {
-                runCallbacks(operationCallbacks, c -> c.onError(context, operationName, query, variables, data, errors));
+                runCallbacks(operationCallbacks, c -> c.onError(context, operationName, query, variables, data, errors, extensions));
             } else {
-                runCallbacks(operationCallbacks, c -> c.onSuccess(context, operationName, query, variables, data));
+                runCallbacks(operationCallbacks, c -> c.onSuccess(context, operationName, query, variables, data, extensions));
             }
 
-            runCallbacks(operationCallbacks, c -> c.onFinally(context, operationName, query, variables, data));
+            runCallbacks(operationCallbacks, c -> c.onFinally(context, operationName, query, variables, data, extensions));
         }
     }
 
-    private Map<String, Object> createResultFromDataAndErrors(Object data, List<GraphQLError> errors) {
+    private Map<String, Object> createResultFromDataErrorsAndExtensions(Object data, List<GraphQLError> errors, Object extensions) {
 
-        final Map<String, Object> result = new HashMap<>();
```

Comment:
What's the reasoning behind changing this to a linked map?  Are the responses ordered?

Reply:
Pretty much.  This ensures that the response has "data", "errors", "extensions" in that order.  More of convenience/usability thing, especially since the extensions can contain quite a bit of data.